main/guides/coreeval/local-testnet.md

```diff
@@ -1,45 +1,187 @@
+For a local agoric blockchain:
 
-	```jsx
-	agoric init <The name of your project>
-	```
-    
-    For example, if you wanted to run your local blockchain from a folder named "Demo", you'd run this command:
-    
-    ```jsx
-    agoric init Demo
-    ```
+1. Ensure that you have [started your local agoric blockchain](../getting-started/#starting-a-local-agoric-blockchain):
+use `yarn docker:logs` to check that your chain is running; if not, use `yarn start:docker` to start it.
 
-	**Note:** Your project folder should *not* be located within your local clone of the agoric SDK.
+2. Build the transaction contents with `cd contract; yarn build:deployer`; _details below_.
 
-3. Install additional dependencies by entering your project folder and running the following command.
+3. Sign and broadcast the transactions with `yarn start`; _details below_.
 
-	```jsx
-    cd <Name of your project (e.g., Demo)>
-    agoric install
-    ```
+For testnet access and governance norms, see [Guide to Agoric Testnets \- Agoric Community Forum](https://community.agoric.com/t/guide-to-agoric-testnets/515).
 
-4.  Start the chain by running the following command. (**Note:** You should still be located within your project folder.)
+_Mainnet deployment is out of scope of this document; see [Mainnet 2 Deployment Checklist #8280](https://github.com/Agoric/agoric-sdk/discussions/8280) in agoric-sdk discussions._
 
-	```jsx
-	agoric start local-chain --verbose --reset
-	```
+## Bundling Core Eval Scripts with `build:deployer` and `rollup.config.mjs`
 
-5. Wait for the first block to be produced.
-6. Open a second terminal.
-7. Within the second terminal, navigate to `<agoric-sdk>/bin` and submit the governance proposal by running the following command. (Make sure to enter "y" when asked to confirm the transaction.)
+Core eval scripts are evaluated, not loaded as modules; so any `import` or `export` declarations are syntax errors. In **dapp-agoric-basics**,  we use [rollup](https://rollupjs.org/) to support developing
+a core-eval script and permit as a module:
 
-	```
-    ./agd --chain-id=agoriclocal --title=<Insert your own title> --description=<Insert your description> --home=<PATH to your project folder>/_agstate/keys --keyring-backend=test --from=provision tx gov submit-proposal swingset-core-eval <PATH to permissions> <PATH to proposal>
-	```
+   - `import { E } from '@endo/far'`
+     A bundle strips this declaration during bundling
+     since the core-eval scope includes exports of `@endo/far`.
+   - `bundleID = ...` is replaced using updated/cached bundle hash
+   - `main` is appended as the script completion value
+   - the `permit` export is emitted to a `.json` file
 
-    For example, to deploy the PSM proposal referenced in the previous topics, run the following:
+See [rollup.config.mjs](https://github.com/Agoric/dapp-agoric-basics/blob/dc-vote-fee/contract/rollup.config.mjs) for details on using the
```

Comment:
this is a link to the `dc-vote-fee` branch, which is WIP

Reply:
I plan to land `dc-vote-fee` first and change the link here before landing this one.